commons-numbers-complex-streams/src/main/java/org/apache/commons/numbers/complex/streams/ComplexUtils.java

```diff
@@ -651,79 +726,71 @@ public static Complex extractComplexFromInterleavedArray(float[] f, int index) {
 
     /**
      * Converts a complex interleaved {@code double[]} array to a
-     * {@code Complex[]} array
+     * {@code Complex[]} array.
      *
      * @param interleaved array of numbers to be converted to their {@code Complex} equivalent
      * @return {@code Complex} array
-     *
-     * @since 4.0
      */
     public static Complex[] interleaved2Complex(double[] interleaved) {
         final int length = interleaved.length / 2;
-        final Complex c[] = new Complex[length];
+        final Complex[] c = new Complex[length];
         for (int n = 0; n < length; n++) {
-            c[n] = new Complex(interleaved[n * 2], interleaved[n * 2 + 1]);
+            c[n] = Complex.ofCartesian(interleaved[n * 2], interleaved[n * 2 + 1]);
         }
         return c;
     }
 
     /**
      * Converts a complex interleaved {@code float[]} array to a
-     * {@code Complex[]} array
+     * {@code Complex[]} array.
      *
      * @param interleaved float[] array of numbers to be converted to their {@code Complex} equivalent
      * @return {@code Complex} array
-     *
-     * @since 4.0
      */
     public static Complex[] interleaved2Complex(float[] interleaved) {
         final int length = interleaved.length / 2;
-        final Complex c[] = new Complex[length];
+        final Complex[] c = new Complex[length];
         for (int n = 0; n < length; n++) {
-            c[n] = new Complex(interleaved[n * 2], interleaved[n * 2 + 1]);
+            c[n] = Complex.ofCartesian(interleaved[n * 2], interleaved[n * 2 + 1]);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).